pkg/provisioner/resourcehandlers.go

```diff
@@ -176,6 +225,106 @@ func createConfigMap(obc *v1alpha1.ObjectBucketClaim, ep *v1alpha1.Endpoint, lab
 	return configMap, err
 }
 
+func getExistingSecret(obc *v1alpha1.ObjectBucketClaim, c kubernetes.Interface, retryInterval, retryTimeout time.Duration) (secret *corev1.Secret, err error) {
+	err = wait.PollImmediate(retryInterval, retryTimeout, func() (done bool, err error) {
```

Comment:
I'm not sure we need polling here. The controller will retry when an error is returned to some max number of attempts. So, I think it's ok to not poll, return an error, even if it's an intermittent error and retry naturally? But, I have not gone through all the code to check idempotency.

Reply:
I do agree I think. It's a matter of whether polling/recovery is better in the face of random API errors or erroring out to go through the provisioning loop again. I suppose it's most important to handle the random errors when creating/updating resources IMO. A pro of having a polling loop in functions even like this does mean that the provision function can be a more stable atomic unit, and I do see polling loops used a lot throughout the code as it stands already.

Reply:
Agree we used polling throughout the lib code (but, unfortunately, not consistently. We may have not understood controllers well enough back then? I do agree that these polls make the func more "atomic" and likely hide all intermittend k8s api errors. Still, it's extra code and slightly impairs readability... Probably fine in this PR to continue the lib's use of Poll and, if it ever matters, handle it in a separate PR.